site/app/controllers/HomePageController.php

```diff
@@ -101,25 +102,64 @@ public function getGroups($user_id = null): MultiResponse {
      * @return MultiResponse
      */
     public function showHomepage() {
+        $user = $this->core->getUser();
         $courses = $this->getCourses()->json_response->json;
 
+
+
         return new MultiResponse(
             null,
             new WebResponse(
                 ['HomePage'],
                 'showHomePage',
                 $this->core->getUser(),
+                $this->core->getCsrfToken(),
                 $courses["data"]["unarchived_courses"],
                 $courses["data"]["archived_courses"]
             )
         );
     }
 
+    /**
+     * @Route("/home/get_user_time_zone", methods={"GET"})
+     * @return JsonResponse
+     *
+     * Handle ajax request to update the currently logged in user's time zone data.
+     *
+     * Will return a json success or failure response depending on the result of the operation.
+     */
+    public function getUserTimeZone() {
+         $user = $this->core->getUser();
+         $time_zone = $this->core->getQueries()->getSubmittyUserTimeZone($user);
+
+
+            // Updating went smoothly, so return success
+        if ($time_zone != null) {
+            $offset = DateUtils::getUTCOffset($time_zone);
+            $user_time_zone_with_offset = "(UTC" . $offset . ") " . $time_zone;
+            return JsonResponse::getSuccessResponse([
+                'utc_offset' => $offset,
+                'user_time_zone_with_offset' => $user_time_zone_with_offset
+            ]);
+        }
+
+
+        // Some failure occurred
+        return JsonResponse::getFailResponse('Error encountered getting user time zone.');
+    }
+
+
+
+
+
+
     /**
      * @Route("/home/courses/new", methods={"POST"})
      * @Route("/api/courses", methods={"POST"})
      */
     public function createCourse() {
+
+
```

Comment:
This whitespace change is unnecessary.  (the same comment applies elsewhere in this file)